h/accounts/test/models_test.py

```diff
@@ -132,3 +134,61 @@ def test_staff_members_does_not_return_non_staff_users():
 
     for non_staff in non_staff:
         assert non_staff not in staff
+
+
+activate_fixtures = pytest.mark.usefixtures('events')
+
+
+def mock_request():
+    """Return a mock request object for User.activate() tests."""
+    class DBSpec(object):
+        def delete(self, activation):
+            pass
+
+    request = testing.DummyRequest(db=mock.Mock(spec=DBSpec))
+
+    def notify_spec(event):
+        pass
+
+    request.registry.notify = mock.Mock(spec=notify_spec)
+
+    return request
+
+
+@activate_fixtures
+def test_User_activate_deletes_activation():
+    request = mock_request()
+    user = factories.User()
+
+    user.activate(request)
+
+    request.db.delete.assert_called_once_with(user.activation)
```

Comment:
I'd suggest not attempting to mock out the database session in model tests. Just use the database. Set up some data, run some functions, and then verify the state of the database.